File: client_android/app/src/main/java/com/securecall/app/ghostnet/frames/AudioFrame.java
package com.securecall.app.ghostnet.frames;

public class AudioFrame {

    private final byte[] data;

    public AudioFrame(byte[] data) {
        this.data = data;
    }

    public byte[] getData() {
        return data;
    }

    public GhostFrame toGhostFrame() {
        return new GhostFrame(GhostFrame.FrameType.AUDIO, data);
    }
}

File: client_android/app/src/main/java/com/securecall/app/ghostnet/transport/frame/AudioFrame.kt
package com.securecall.app.ghostnet.transport.frame

// BACKEND-38 / ANDROID-03:
// Placeholder AudioFrame — Grundstruktur für zukünftige Audio-Pipeline
class AudioFrame(
    val timestamp: Long,
    val type: FrameType,
    val data: ByteArray
) {

    enum class FrameType {
        RAW,        // ungefiltertes PCM oder RTP-Paket
        ENCRYPTED,  // SRTP / Noise-based encryption
        CONTROL     // z.B. DTX, Silence, Heartbeat
    }

    override fun toString(): String {
        return "AudioFrame(ts=$timestamp, type=$type, size=${data.size})"
    }
}
